src/Stratis.Bitcoin.Features.SmartContracts/Rules/SmartContractCoinViewRuleLogic.cs

```diff
@@ -64,15 +68,20 @@ public async Task RunAsync(Func<RuleContext, Task> baseRunAsync, RuleContext con
             this.refundCounter = 1;
 
             Block block = context.ValidationContext.BlockToValidate;
+            this.logger.LogDebug("Block to validate '{0}'", block.GetHash());
 
             // Get a IStateRepositoryRoot we can alter without affecting the injected one which is used elsewhere.
-            byte[] blockRoot = ((ISmartContractBlockHeader)context.ValidationContext.ChainedHeaderToValidate.Previous.Header).HashStateRoot.ToBytes();
-            this.mutableStateRepository = this.stateRepositoryRoot.GetSnapshotTo(blockRoot);
+            uint256 blockRoot = ((ISmartContractBlockHeader)context.ValidationContext.ChainedHeaderToValidate.Previous.Header).HashStateRoot;
+
+            this.logger.LogDebug("Block hash state root '{0}'", blockRoot);
+            byte[] blockRootBytes = blockRoot.ToBytes();
+            this.mutableStateRepository = this.stateRepositoryRoot.GetSnapshotTo(blockRootBytes);
 
             await baseRunAsync(context);
 
             var blockHeader = (ISmartContractBlockHeader)block.Header;
 
+            this.logger.LogDebug("Compare state roots '{0}' and '{1}'", new uint256(this.mutableStateRepository.Root), blockHeader.HashStateRoot);
```

Comment:
Might make sense to break out `Root` and `HashStateRoot` into variables too, and then use the same variables when logging and in the comparison below. Otherwise their values could change in between logging and performing the comparison.